### ISSUE TITLE
[BUG] dataset.unique_ids returns an empty set

### ISSUE THREAD

**Issue by Bloodrammer** [voxel51/fiftyone#2895]
### System information

-   **OS Platform and Distribution** (e.g., Linux Ubuntu 16.04): Linux Ubuntu 20.04
-   **Python version** (`python --version`): 3.8.13
-   **FiftyOne version** (`fiftyone --version`): 0.20.1
-   **FiftyOne installed from** (pip or source): pip

### Commands to reproduce

https://docs.voxel51.com/user_guide/brain.html#cifar-10-example

### Describe the problem

I tried calling `find_unique(count)` on my data and it returned the following output:

```
Computing unique samples...
threshold: -, kept: 500, target: 500
Uniqueness computation complete
```

Then I called `image_index.unique_ids` and it returned `set()`. By the way, [https://docs.voxel51.com/user_guide/brain.html#finding-maximally-unique-images](this) documentation snippet suggests that `unique_ids` used to be a list, so it needs to be updated. 

The same issue occurs with the CIFAR example. 

### Code to reproduce issue

The CIFAR example from the documentation.

### Other info/logs

-

### What areas of FiftyOne does this bug affect?

-   [ ] `App`: FiftyOne application issue
-   [x] `Core`: Core Python library issue
-   [ ] `Server`: FiftyOne server issue

### Willingness to contribute

The FiftyOne Community encourages bug fix contributions. Would you or another member of your organization be willing to contribute a fix for this bug to the FiftyOne codebase?

-   [ ] Yes. I can contribute a fix for this bug independently
-   [ ] Yes. I would be willing to contribute a fix for this bug with guidance from the FiftyOne community
-   [x] No. I cannot contribute a bug fix at this time

**Comment by nnkwame** [comment]
I have encountered the same issue with `find_unique(count)`.

Something to note: It is reproducible even with `count` as low as 2. This suggests that the `ids` list passed to the underlying call is empty/ not populated correctly, looking at this snippet from the underlying call:

```python
...
def _remove_duplicates_count(self, num_keep, ids, init_thresh=None):
...
    if num_keep <= 0:
        logger.info(
            "threshold: -, kept: %d, target: %d", num_keep, num_keep
        )
        return set(), None

    if num_keep >= len(ids):
        logger.info(
            "threshold: -, kept: %d, target: %d", num_keep, num_keep
        )
        return set(ids), None
...
```

In my case I am running `compute_similarity()` on patches, following the example in the Object Similarity tab [here](https://docs.voxel51.com/user_guide/brain.html#precomputed-embeddings).

The value of `patches_field` seems to decide what ids are passed to `_remove_duplicates_count()`. 

Maybe the population of `_current_label_ids` during Similarity-Indexing is bugged?



fiftyone version:
```
fiftyone                    0.20.0
fiftyone-brain              0.11.0
fiftyone-db                 0.4.0
python                    3.10.8 
```

**Comment by nnkwame** [comment]
Tested the population of SimilarityIndex object resulting from  `compute_similarity()`


the following:
```python
... #compute embeddings ..
...
results = fob.compute_similarity(
    my_dataset,
    patches_field="ground_truth",
    embeddings=patch_embeddings,
    brain_key="gt_sim",
)
print('sample ids of osc:')
print(f'current: {results.current_sample_ids}')
print(f'general: {results.sample_ids}')
print('label ids of osc:')
print(f'current: {results.current_label_ids}')
print(f'general: {results.label_ids}')

```

has this output on 0.20.0:

```
sample ids of osc:
current: []
general: ['645c1e3925d3b78c2438ecb6' '645c1e3925d3b78c2438ecb9'
 '645c1e3925d3b78c2438ecbc' ... '645c1e3f25d3b78c24393b11'
 '645c1e3f25d3b78c24393b12' '645c1e3f25d3b78c24393b13']
label ids of osc:
current: []
general: ['645c1e3925d3b78c2438ecb5' '645c1e3925d3b78c2438ecb7'

```

and this on 0.19.0:

```
sample ids of osc:
current: ['645c181aa9931ad186dc98ce' '645c181aa9931ad186dc98d1'
 '645c181aa9931ad186dc98d4' ... '645c1820a9931ad186dce729'
 '645c1820a9931ad186dce72a' '645c1820a9931ad186dce72b']
general: ['645c181aa9931ad186dc98ce' '645c181aa9931ad186dc98d1'
 '645c181aa9931ad186dc98d4' ... '645c1820a9931ad186dce729'
 '645c1820a9931ad186dce72a' '645c1820a9931ad186dce72b']
label ids of osc:
current: ['645c181aa9931ad186dc98cd' '645c181aa9931ad186dc98cf'
 '645c181aa9931ad186dc98d2' ... '645c1820a9931ad186dce5b7'
 '645c1820a9931ad186dce5b8' '645c1820a9931ad186dce5b9']
general: ['645c181aa9931ad186dc98cd' '645c181aa9931ad186dc98cf'
 '645c181aa9931ad186dc98d2' ... '645c1820a9931ad186dce5b7'
 '645c1820a9931ad186dce5b8' '645c1820a9931ad186dce5b9']

```

There appears to be an issue with `_current_sample_ids` , `_current_label_ids` in v 0.20

**Comment by brimoor** [comment]
Thanks for catching this, I was able to reproduce in `fiftyone==0.20`. We'll have it patched in `fiftyone==0.21`.

In the meantime, you can manually patch the bug by inserting `use_view()` after creating your similarity index:

```py
results = fob.compute_similarity(dataset, ...)
results.use_view(dataset)
```

**Comment by brimoor** [comment]
> @brimoor can I call `reload()` on an already existing SimilarityIndex stored in a persistent dataset?

yes it's harmless to run. In fact, it's only needed when you create a new index or call `add_to_index()` within a Python session. If you're working with a persistent dataset, if you start a new session and load the results, no `reload()` will be necessary:

```py
import fiftyone as fo

dataset = fo.load_dataset(...)

# No `reload()` necessary when loading an existing index
results = dataset.load_brain_results(...)
```
